packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -368,15 +369,26 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
   }
 
   double getPageFromPixels(double pixels, double viewportDimension) {
-    return math.max(0.0, pixels) / math.max(1.0, viewportDimension * viewportFraction);
+    final double actual = math.max(0.0, pixels) / math.max(1.0, viewportDimension * viewportFraction);
+    final int round = actual.round();
+    if ((actual - round).abs() < precisionErrorTolerance) {
+      return round.toDouble();
+    }
+    return actual;
   }
 
   double getPixelsFromPage(double page) {
     return page * viewportDimension * viewportFraction;
   }
 
   @override
-  double get page => pixels == null ? null : getPageFromPixels(pixels.clamp(minScrollExtent, maxScrollExtent), viewportDimension);
+  double get page {
+    assert(
```

Comment:
Is this assert correct? Seems like if pixels is null we don't care whether minScrollExtent or maxScrollExtent is non-null?

Reply:
no, I missed the pixel is nullable. fixed